prometheus/indico_prometheus/metrics.py

```diff
@@ -0,0 +1,150 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2023 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from datetime import timedelta
+
+from prometheus_client import metrics
+
+from indico.core.db import db
+from indico.modules.attachments.models.attachments import Attachment, AttachmentFile
+from indico.modules.auth.models.identities import Identity
+from indico.modules.categories.models.categories import Category
+from indico.modules.events.models.events import Event
+from indico.modules.rb.models.reservation_occurrences import ReservationOccurrence
+from indico.modules.rb.models.reservations import Reservation
+from indico.modules.rb.models.rooms import Room
+from indico.modules.users.models.users import User
+from indico.util.date_time import now_utc
+
+from indico_prometheus.queries import get_attachment_query, get_note_query
+
+
+# Check for availability of the LiveSync plugin
+LIVESYNC_AVAILABLE = True
+try:
+    from indico_livesync.models.queue import LiveSyncQueueEntry
+except ImportError:
+    LIVESYNC_AVAILABLE = False
+
+
+num_active_events = metrics.Gauge('indico_num_active_events', 'Number of Active Events')
+num_events = metrics.Gauge('indico_num_events', 'Number of Events')
+
+num_active_users = metrics.Gauge('indico_num_active_users', 'Number of Active Users (logged in in the last 24h)')
+num_users = metrics.Gauge('indico_num_users', 'Number of Users')
+
+num_categories = metrics.Gauge('indico_num_categories', 'Number of Categories')
+
+num_active_attachment_files = metrics.Gauge('indico_num_active_attachment_files', 'Number of attachment files')
+num_attachment_files = metrics.Gauge(
+    'indico_num_attachment_files',
+    'Total number of attachment files, including older versions / deleted'
+)
+
+size_active_attachment_files = metrics.Gauge(
+    'indico_size_active_attachment_files',
+    'Total size of all active attachment files (bytes)'
+)
+size_attachment_files = metrics.Gauge(
+    'indico_size_attachment_files',
+    'Total size of all attachment files, including older versions / deleted (bytes)'
+)
+
+num_notes = metrics.Gauge('indico_num_notes', 'Number of notes')
+
+num_active_rooms = metrics.Gauge('indico_num_active_rooms', 'Number of active rooms')
+num_rooms = metrics.Gauge('indico_num_rooms', 'Number of rooms')
```

Comment:
"Number of restricted rooms" could also be an interesting metric (number of rooms with `Room.is_protected`)

Reply:
And "Number of rooms requiring manual booking confirmation" (`reservations_need_confirmation`)

Reply:
Got it.